src/credentials/CHIPCert.cpp

```diff
@@ -667,61 +629,150 @@ void ValidationContext::Reset()
     mRequiredCertType = kCertType_NotSpecified;
 }
 
-CHIP_ERROR DetermineCertType(ChipCertificateData & cert)
+bool ChipRDN::IsEqual(const ChipRDN & other) const
 {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-
-    // If the certificate subject contains a ChipCAId attribute...
-    if (cert.mSubjectDN.mAttrOID == kOID_AttributeType_ChipCAId)
+    if (mAttrOID == kOID_Unknown || mAttrOID == kOID_NotSpecified || mAttrOID != other.mAttrOID)
     {
-        // Verify the BasicConstraints isCA flag is true.
-        VerifyOrExit(cert.mCertFlags.Has(CertFlags::kIsCA), err = CHIP_ERROR_CERT_USAGE_NOT_ALLOWED);
-
-        // Verify the key usage extension is present and contains the 'keyCertSign' flag.
-        VerifyOrExit(cert.mCertFlags.Has(CertFlags::kExtPresent_KeyUsage) && cert.mKeyUsageFlags.Has(KeyUsageFlags::kKeyCertSign),
-                     err = CHIP_ERROR_CERT_USAGE_NOT_ALLOWED);
-
-        // Set the certificate type to CA.
-        cert.mCertType = kCertType_CA;
+        return false;
     }
 
-    // If the certificate subject contains a ChipNodeId attribute set the certificate type to Node.
-    else if (cert.mSubjectDN.mAttrOID == kOID_AttributeType_ChipNodeId)
+    if (IsChipIdX509Attr(mAttrOID))
+    {
+        return mAttrValue.mChipId == other.mAttrValue.mChipId;
+    }
+    else
     {
-        cert.mCertType = kCertType_Node;
+        return (mAttrValue.mString.mLen == other.mAttrValue.mString.mLen &&
+                memcmp(mAttrValue.mString.mValue, other.mAttrValue.mString.mValue, mAttrValue.mString.mLen) == 0);
     }
+}
+
+ChipDN::ChipDN()
+{
+    Clear();
+}
+
+ChipDN::~ChipDN()
+{
+    Clear();
```

Comment:
Why do we need to manually clear in the destructor?

Likewise, is there a reason we need to manually clear this memory in the constructor?

Reply:
removed manual Clear() call from constructor and destructor methods.